### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by LtSnail**

### CODE REVIEW

File: Engine/CMakeLists.txt [hunk 1]
```diff
@@ -64,8 +65,8 @@ elseif ( WIN32 )
 
     else()
         message(FATAL_ERROR "Target platform is not X64 - Compilation denied")
-    endif() # End of Windows
-endif() # End of MAC
```
Comment (Mon1Amour4) [inline Engine/CMakeLists.txt:68]: I thought it might be useful, a little bit more information :)

Comment (LtSnail) [reply Engine/CMakeLists.txt:68]: Then we need to add this to each if statement eventually that might add clutter

File: Engine/Fuego/Application.h [hunk 2]
```diff
@@ -11,10 +11,11 @@ namespace Fuego
     class Application
     {
     public:
-       FUEGO_API Application();
-       FUEGO_API virtual ~Application();
+		FUEGO_API Application();
+        FUEGO_API virtual ~Application() = default;
+        FUEGO_NON_COPYABLE_NON_MOVABLE(Application)
```
Comment (Mon1Amour4) [inline Engine/Fuego/Application.h:17]: Why do we need FUEGO_NON_COPYABLE_NON_MOVABLE? Does it relate to .dll export\import issue ?

File: Engine/Fuego/EventQueue.h [hunk 3]
```diff
@@ -9,6 +9,9 @@ namespace Fuego
     class FUEGO_API EventQueue
     {
     public:
+        EventQueue() = default;
+        FUEGO_NON_COPYABLE_NON_MOVABLE(EventQueue)
+
```
Comment (Mon1Amour4) [inline Engine/Fuego/EventQueue.h:14]: Why do we need FUEGO_NON_COPYABLE_NON_MOVABLE? Does it relate to .dll export\import issue ?

File: Engine/CMakeLists.txt (hunk 1 above)
Comment (Mon1Amour4) [reply Engine/CMakeLists.txt:68]: Fair enough

File: Engine/Fuego/Application.h (hunk 2 above)
Comment (LtSnail) [reply Engine/Fuego/Application.h:17]: No, it is just the way to comply with the [Rule of Five](https://en.cppreference.com/w/cpp/language/rule_of_three).

Comment (LtSnail) [reply Engine/Fuego/Application.h:17]: I used clang static analyser to find all those issues

File: Engine/Fuego/EventQueue.h (hunk 3 above)
Comment (LtSnail) [reply Engine/Fuego/EventQueue.h:14]: If a class will not be moved or copied it's better to delete related operators and constructors

**Pull request merged**